config/lib/seed.js

```diff
@@ -2,93 +2,145 @@
 
 var mongoose = require('mongoose'),
   chalk = require('chalk'),
-  crypto = require('crypto'),
-  User = mongoose.model('User');
+  crypto = require('crypto');
 
-console.log(chalk.bold.red('Warning:  Database seeding is turned on'));
-
-var seedUser = {
-  username: 'user',
-  password: 'User_Password1!',
-  provider: 'local',
-  email: 'user@localhost.com',
-  firstName: 'User',
-  lastName: 'Local',
-  displayName: 'User Local',
-  roles: ['user']
-};
-
-var seedAdmin = {
-  username: 'admin',
-  password: 'Admin_Password1!',
-  provider: 'local',
-  email: 'admin@localhost.com',
-  firstName: 'Admin',
-  lastName: 'Local',
-  displayName: 'Admin Local',
-  roles: ['user', 'admin']
-};
-
-//If production only seed admin if it does not exist
-if (process.env.NODE_ENV === 'production') {
-  //Add Local Admin
-  User.find({username: seedAdmin.username}, function (err, users) {
-    if (users.length === 0) {
-      var user = new User(seedAdmin);
-
-      // generate a random password and save 
-      User.generateRandomPassphrase()
-      .then(saveUser(user))
-      .catch(reportError);
-
-    } else {
-      console.log(seedAdmin.username + ' user exists');
-    }
+function removeUser (user) {
+  return new Promise(function (resolve, reject) {
+    var User = mongoose.model('User');
+    User.find({username: user.username}).remove(function (err) {
+        if (err) {
+          reject(new Error('Database Seeding:\t\t\tFailed to remove local ' + user.username));
+        }
+        resolve();
+    });
   });
-} else {
-
-  //Add Local User
-  User.find({username: seedUser.username}).remove(function () {
-    var user = new User(seedUser);
+}
 
-    // generate a random password and save
-    User.generateRandomPassphrase()
-    .then(saveUser(user))
-    .catch(reportError);
-  });
+function saveUser (user) {
+  return function() {
```

Comment:
@lirantal I see what you mean. However, I think this behavior is stemming from the previous promise resolve in the chain. In both uses of the `saveUser` method, the previous promise is resolving with no data: `resolve();`

Hence, the `saveUser` method is expecting to handle that resolve, but if you don't have that closure there it expects that your `user` parameter of `saveUser` is coming from the resolve (which would be *undefined*).

Let's say you resolved `removeUser` with data: `resolve(user);` 
The next operation in the chain could look like this `.then(saveUser)`; in this case you wouldn't need that empty closure since it would get the `user` from the resolve.

Ok... Now with all that said. I think what you have is the correct way to handle it. I just wanted to clarify what my findings were, as to why we're seeing this behavior & need for the empty closure; and I'm sure you're aware of most of this. Furthermore, I think you're absolutely correct to resolve most of those functions with no data. For instance, it wouldn't make sense to resolve `removeUser`, and we want to maintain the `user` object from the scope that the chain is in. 

I'm very pleased with how this turned out! Thanks for working through this with me. I've definitely learned a few things. I think this is ready.